### DISCUSSION TITLE
Blocks can be selected without a blue overlay. How to make the overlay to appear?

### DISCUSSION THREAD

**Discussion by goliney** [xyflow/xyflow#4254] [General]
I'm talking about `<NodesSelection />` that is controlled by `nodesSelectionActive`.

If I drag with <kbd>Shift</kbd> pressed, the nodes selection appears. But if I select blocks with <kbd>Meta</kbd> pressed, the overlay doesn't appear.

I want to always have the blue overlay if more than one block is selected.

❓ Is it safe to use this:

```typescript
const { setState } = useStoreApi();
setState({ nodesSelectionActive: true }); // show blue overlay
```

[Codesandbox](https://codesandbox.io/p/sandbox/upbeat-yonath-jhd92g)

https://github.com/xyflow/xyflow/assets/2017344/ba72f6f6-ad68-47d5-9dfb-ef04cf7223f8

**Comment by peterkogo** [comment]
This is a legacy behavior we would like to change in the next major version. It makes no sense that there are two tiers of selections.
